src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/DirectoryBasedStoreProvider.cs

```diff
@@ -0,0 +1,400 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+
+namespace Internal.Cryptography.Pal
+{
+    /// <summary>
+    /// Provides an implementation of an X509Store which is backed by files in a directory.
+    /// </summary>
+    internal class DirectoryBasedStoreProvider : IStorePal
+    {
+        // {thumbprint}.1.pfx to {thumbprint}.9.pfx
+        private const int MaxSaveAttempts = 9; 
```

Comment:
I don't like the idea of `Add` having a theoretically infinite loop due to someone poisoning the directory.

Since the base portion of the filename is the SHA-1 hash of the certificate (aka the thumbprint/fingerprint) the chance of needing to use this limiter at all is low.  So this is allowing for 10 different certificates all with the same thumbprint (or up to 10 poison files) before we give up and tell someone that the universe stopped making sense.

And I'll admit that I went with 9 because it allowed for a nice pattern of {stem}.pfx and {stem}.1.pfx to {stem}.9.pfx in decimal formatting.